pkgs/tools/networking/hostname-debian/default.nix

```diff
@@ -0,0 +1,28 @@
+{ stdenv, lib, fetchurl }:
+
+stdenv.mkDerivation rec {
+  pname = "hostname-debian";
+  version = "3.23";
+
+  src = fetchurl {
+    url = "https://deb.debian.org/debian/pool/main/h/hostname/hostname_${version}.tar.gz";
+    sha256 = "sha256-vG0ZVLIoSYaf+LKmAuOfCLFwL2htS1jdeSfN61tIdu8=";
+  };
+
+  prePatch = ''
```

Comment:
```suggestion
  postPatch = ''
```

Reply:
Can you explain your preference for postPatch?

grep -rC4 -- '-o root' pkgs
shows lots of places with this type of patch. Some use a separate patch file, otherwise it's:
- prePatch (e.g. super, pptp, x2goserver)
- postPatch (datefudge, ipad_charge, pptpd)
- postConfigure (pmount, bsdgames)
- preBuild (crunch, dd_rescue, cron)
- patchPhase (gt5, pal, pam_ccreds)

I can't see a clear winner.

My thought was that prePatch will apply to unmodified sources, while postPatch will have to deal with sources that may have been modified. But since patchPhase is a nop here, prePatch and postPatch are equivalent.

Reply:
If there is a patch that touches the surrounding lines then it wouldn't apply any longer and it is not my personal preference but a general rule.

> patchPhase (gt5, pal, pam_ccreds)

That is completely wrong and breaks patches.

> I can't see a clear winner.

You're methodology is not very good since it also accounts for old packages that are not up to date with current standards.